SFS_time_domain/driving_function_imp_nfchoa.m

```diff
@@ -129,24 +132,38 @@
     end
 end
 
-% -------------------------------------------------------------------------
-%                      ___
-%                1     \
-% D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
-%             2 pi r0  m=-M..M
-%
-% see Spors et al. (2011), eq.(4) and (5)
-%--------------------------------------------------------------------------
+if strcmp('2.5D',dimension)
+    % -------------------------------------------------------------------------
+    %                      ___
+    %                1     \
+    % D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
+    %             2 pi r0  m=-M..M
+    %
+    % see Spors et al. (2011), eq.(4) and (5)
+    %--------------------------------------------------------------------------
+    
+    % apply modal weighting
+    wm = modal_weighting(order,conf);  % modal weighting function
+    Hm = bsxfun(@times,Hm,wm);
```

Comment:
Well it could be the case for other windows, but I'm not sure. It really depends on the optimisation criterion, which is fulfilled by a specific window function. I will include the 3D case inside ``modal_weighting``.